utils/global-state-update-gen/src/generic.rs

```diff
@@ -0,0 +1,284 @@
+pub(crate) mod config;
+mod state_reader;
+mod state_tracker;
+#[cfg(test)]
+mod testing;
+
+use std::{
+    collections::{BTreeMap, BTreeSet},
+    fs,
+};
+
+use casper_engine_test_support::LmdbWasmTestBuilder;
+use casper_types::{
+    system::auction::{Bid, SeigniorageRecipient, SeigniorageRecipientsSnapshot},
+    CLValue, EraId, Key, PublicKey, StoredValue, U512,
+};
+
+use clap::ArgMatches;
+
+use crate::utils::{hash_from_str, print_entry, validators_diff, ValidatorsDiff};
+
+use self::{
+    config::{AccountConfig, Config, Transfer},
+    state_reader::StateReader,
+    state_tracker::StateTracker,
+};
+
+pub(crate) fn generate_generic_update(matches: &ArgMatches<'_>) {
+    let data_dir = matches.value_of("data_dir").unwrap_or(".");
+    let state_hash = hash_from_str(matches.value_of("hash").unwrap());
+    let config_path = matches.value_of("config_file").unwrap();
+
+    let config_bytes = fs::read(config_path).expect("couldn't read the config file");
+    let config: Config = toml::from_slice(&config_bytes).expect("couldn't parse the config file");
+
+    let builder = LmdbWasmTestBuilder::open_raw(data_dir, Default::default(), state_hash);
+
+    update_from_config(builder, config);
+}
+
+pub(crate) fn get_update<T: StateReader>(reader: T, config: Config) -> BTreeMap<Key, StoredValue> {
+    let mut state_tracker = StateTracker::new(reader);
+
+    process_transfers(&mut state_tracker, &config.transfers);
```

Comment:
So `process_transfers` does what the command `balances` does, and `update_account_balances`/`update_auction_state` does what `validators` does?

Reply:
Pretty much, yes :+1: